server/server.go

```diff
@@ -773,6 +773,38 @@ func (j *jsonRPCHub) GetIteratorDumpTree(block *types.Block, opts *state.DumpInf
 	return itDump, nil
 }
 
+// StorageRangeAt returns the storage at the given block height and transaction index.
+func (j *jsonRPCHub) StorageRangeAt(storageRangeResult *state.StorageRangeResult, block *types.Block,
+	addr *types.Address, keyStart []byte, txIndex, maxResult int) error {
+	if block.Number() == 0 {
+		return fmt.Errorf("genesis block can't have transaction")
+	}
+
+	if txIndex < 0 || txIndex >= len(block.Transactions) {
+		return fmt.Errorf("transaction index %d out of bounds, block contains %d transactions",
+			txIndex, len(block.Transactions))
+	}
+
+	parentHeader, ok := j.GetHeaderByHash(block.ParentHash())
+	if !ok {
+		return fmt.Errorf("parent header not found for block %s", block.ParentHash().String())
+	}
+
+	blockProposer := types.BytesToAddress(block.Header.Miner)
+
+	transition, err := j.BeginTxn(parentHeader.StateRoot, block.Header, blockProposer)
+	if err != nil {
+		return fmt.Errorf("failed to begin transaction: %w", err)
+	}
+
+	tx := block.Transactions[txIndex]
+	if _, err := transition.Apply(tx); err != nil {
```

Comment:
IMO we should execute all transactions with index <= txIndex in order to reach requested state (like we did in TraceTxn for instance).